rust/operator-binary/src/controller.rs

```diff
@@ -1161,3 +1022,109 @@ fn liveness_probe(trino: &TrinoCluster) -> Probe {
         ..Probe::default()
     }
 }
+
+fn create_tls_volume(volume_name: &str, tls_secret_class: Option<&TlsSecretClass>) -> Volume {
+    let secret_class_name = tls_secret_class
+        .map(|t| t.secret_class.as_str())
+        .unwrap_or(TLS_DEFAULT_SECRET_CLASS);
+    VolumeBuilder::new(volume_name)
+        .ephemeral(
+            SecretOperatorVolumeSourceBuilder::new(secret_class_name)
+                .with_pod_scope()
+                .with_node_scope()
+                .build(),
+        )
+        .build()
+}
+
+fn tls_volume_mounts(
+    trino: &TrinoCluster,
+    pod_builder: &mut PodBuilder,
+    cb_prepare: &mut ContainerBuilder,
+    cb_trino: &mut ContainerBuilder,
+    s3_connection: Option<&S3ConnectionSpec>,
+) -> Result<()> {
+    // We always create a mount for tls-certificates (mounted via SecretOperatorVolumeSourceBuilder if tls
+    // is enabled or simply as empty dir to create and copy the system trust store)
+    cb_prepare.add_volume_mount("client-tls-mount", STACKABLE_MOUNT_CLIENT_TLS_DIR);
+    cb_trino.add_volume_mount("client-tls-mount", STACKABLE_MOUNT_CLIENT_TLS_DIR);
+
+    cb_prepare.add_volume_mount("client-tls", STACKABLE_CLIENT_TLS_DIR);
+    cb_trino.add_volume_mount("client-tls", STACKABLE_CLIENT_TLS_DIR);
+    pod_builder.add_volume(
+        VolumeBuilder::new("client-tls")
+            .with_empty_dir(Some(""), None)
+            .build(),
+    );
+    // If tls or authentication are specified we need to provide mounts for certs and keys
+    if trino.tls_enabled() {
+        pod_builder.add_volume(create_tls_volume(
```

Comment:
So you always try to mount this, even if you don't actually specify the volume...?